public/app/features/dashboard-scene/settings/GeneralSettingsEditView.tsx

```diff
@@ -0,0 +1,291 @@
+
+    if (cursorSync instanceof behaviors.CursorSync) {
+      return cursorSync;
+    }
+
+    return;
+  }
+
+  public onTitleChange = (value: string) => {
+    this._dashboard.setState({ title: value });
+  };
+
+  public onDescriptionChange = (value: string) => {
+    this._dashboard.setState({ description: value });
+  };
+
+  public onTagsChange = (value: string[]) => {
+    this._dashboard.setState({ tags: value });
+  };
+
+  public onFolderChange = (newUID: string, newTitle: string) => {
+    const newMeta = {
+      ...this._dashboard.state.meta,
+      folderUid: newUID || this._dashboard.state.meta.folderUid,
+      folderTitle: newTitle || this._dashboard.state.meta.folderTitle,
+      hasUnsavedFolderChange: true,
+    };
+
+    this._dashboard.setState({ meta: newMeta });
+  };
+
+  public onEditableChange = (value: boolean) => {
+    this._dashboard.setState({ editable: value });
+  };
+
+  public onTimeZoneChange = (value: TimeZone) => {
+    this.getTimeRange().setState({
+      timeZone: value,
+    });
+  };
+
+  public onWeekStartChange = (value: string) => {
+    this.getTimeRange().setState({
+      weekStart: value,
+    });
+  };
+
+  public onRefreshIntervalChange = (value: string[]) => {
+    const control = this.getRefreshPicker();
+    control?.setState({
+      intervals: value,
+    });
+  };
+
+  public onNowDelayChange = (value: string) => {
+    // TODO: Figure out how to store nowDelay in Dashboard Scene
+  };
+
+  public onHideTimePickerChange = (value: boolean) => {
+    if (this._dashboard.state.controls instanceof DashboardControls) {
+      for (const control of this._dashboard.state.controls.state.timeControls) {
+        if (control instanceof SceneTimePicker) {
+          control.setState({
+            // TODO: Control visibility from DashboardControls
+            // hidden: value,
+          });
+        }
+      }
+    }
+  };
+
+  public onLiveNowChange = (value: boolean) => {
+    // TODO: Figure out how to store liveNow in Dashboard Scene
+  };
+
+  public onTooltipChange = (value: number) => {
+    this.getCursorSync()?.setState({ sync: value });
+  };
+
+  static Component = ({ model }: SceneComponentProps<GeneralSettingsEditView>) => {
+    const { navModel, pageNav } = useDashboardEditPageNav(model.getDashboard(), model.getUrlKey());
+    const { title, description, tags, meta, editable, overlay } = model.getDashboard().useState();
+    const { sync: graphTooltip } = model.getCursorSync()?.useState() || {};
+    const { timeZone, weekStart } = model.getTimeRange().useState();
+    const { intervals } = model.getRefreshPicker()?.useState() || {};
+
+    const {
+      onTitleChange,
+      onDescriptionChange,
+      onTagsChange,
+      onFolderChange,
+      onEditableChange,
+      onTimeZoneChange,
+      onWeekStartChange,
+      onRefreshIntervalChange,
+      onNowDelayChange,
+      onHideTimePickerChange,
+      onLiveNowChange,
+      onTooltipChange,
+    } = model;
```

Comment:
this seems a little bit unnecessary, reading `onChange={model.onTitleChange}` is more clear as you see right away that it's a callback on the state model, and click to navigate will directly take you to the implementation.

Reply:
@torkelo i agree, but it's way harder to test the model then, as those model methods need to take an event as an input. I'm fine keeping this the way @ivanortegaalba implemented it.

Reply:
@dprokop  not sure I understand the difference in this case, as this is just extracting the methods into local variables, and then passing them in the props. There are no wrapper functions that changes the function signature here. Just feelt these local function variables did not add anything beyond requiring an extra "go to definition" hop to find the real implementation 

But anyway, super nitpit :)

Reply:
But if we do `onChange={model.onTitleChange}` as you suggested we need to change the model method signature.

But yeah, for cases where we do `(e: ChangeEvent<HTMLInputElement>) => onTitleChange(e.target.value)` this could easily be replaced with `(e: ChangeEvent<HTMLInputElement>) => model.onTitleChange(e.target.value)`. I think the event is required for title/description only

Reply:
@dprokop 

I was not suggesting that, just 

```ts
  <TimePickerSettings
    onTimeZoneChange={model.onTimeZoneChange}
    onWeekStartChange={model.onWeekStartChange}
    onRefreshIntervalChange={model.onRefreshIntervalChange}
    onNowDelayChange={model.onNowDelayChange}
    onHideTimePickerChange={model.onHideTimePickerChange}
    onLiveNowChange={model.onLiveNowChange}
```

same for onFolderChange etc, 

for the input handlers, maybe something like 

```ts
<Input
    id="title-input"
    name="title"
    defaultValue={title}
    onBlur={(e: ChangeEvent<HTMLInputElement>) => model.onChangeTitle(e.target.value)}
  />
```

but not a big deal, was just so many lines of code dedicated to kind of pointless variables that only make it a bit hard to navigate to the function

Reply:
I think @torkelo I understood your approach. Is simply to not deconstruct the methods, and just use them through `model.x`. I liked the other way since you can check all the methods you're using from the model at once. Just a matter of personal preference :)